core/src/test/scala/unit/kafka/security/token/TokenManagerTest.scala

```diff
@@ -0,0 +1,344 @@
+
+    assert(tokenManager.getAllTokenInformation().size == 4 )
+
+    //get tokens non-exiting owner
+    var  tokens = getTokens(tokenManager, simpleAclAuthorizer, hostSession, owner1, List(SecurityUtils.parseKafkaPrincipal("User:unknown")))
+    assert(tokens.size == 0)
+
+    //get all tokens for  empty owner list
+    tokens = getTokens(tokenManager, simpleAclAuthorizer, hostSession, owner1, List())
+    assert(tokens.size == 0)
+
+    //get all tokens for owner1
+    tokens = getTokens(tokenManager, simpleAclAuthorizer, hostSession, owner1, List(owner1))
+    assert(tokens.size == 2)
+
+    //get all tokens for owner1
+    tokens = getTokens(tokenManager, simpleAclAuthorizer, hostSession, owner1, null)
+    assert(tokens.size == 2)
+
+    //get all tokens for unknown owner
+    tokens = getTokens(tokenManager, simpleAclAuthorizer, hostSession, SecurityUtils.parseKafkaPrincipal("User:unknown"), null)
+    assert(tokens.size == 0)
+
+    //get all tokens for multiple owners (owner1, renewer4) and without permission for renewer4
+    tokens = getTokens(tokenManager, simpleAclAuthorizer, hostSession, owner1, List(owner1, renewer4))
+    assert(tokens.size == 2)
+
+    //get all tokens for multiple owners (owner1, renewer4) and with permission
+    var acl = new Acl(owner1, Allow, WildCardHost, Describe)
+    simpleAclAuthorizer.addAcls(Set(acl), new Resource(kafka.security.auth.DelegationToken, tokenId3))
+    tokens = getTokens(tokenManager, simpleAclAuthorizer, hostSession, owner1, List(owner1, renewer4))
+    assert(tokens.size == 3)
+
+    //get all tokens for renewer4 which is a renewer principal for some tokens
+    tokens = getTokens(tokenManager, simpleAclAuthorizer, hostSession,  renewer4, List(renewer4))
+    assert(tokens.size == 2)
+
+    //get all tokens for multiple owners (renewer2, renewer3) which are token renewers principals and without permissions for renewer3
+    tokens = getTokens(tokenManager, simpleAclAuthorizer, hostSession,  renewer2, List(renewer2, renewer3))
+    assert(tokens.size == 1)
+
+    //get all tokens for multiple owners (renewer2, renewer3) which are token renewers principals and with permissions
+    hostSession = new Session(renewer2, InetAddress.getByName("192.168.1.1"))
+    acl = new Acl(renewer2, Allow, WildCardHost, Describe)
+    simpleAclAuthorizer.addAcls(Set(acl), new Resource(kafka.security.auth.DelegationToken, tokenId2))
+    tokens = getTokens(tokenManager, simpleAclAuthorizer, hostSession,  renewer2, List(renewer2, renewer3))
+    assert(tokens.size == 2)
+
+    simpleAclAuthorizer.close()
+  }
+
+  def getTokens(tokenManager: TokenManager, simpleAclAuthorizer: SimpleAclAuthorizer, hostSession: Session,
+                requestPrincipal: KafkaPrincipal, requestedOwners: List[KafkaPrincipal]): List[DelegationToken] = {
+
+    if (requestedOwners != null && requestedOwners.isEmpty) {
+      List()
+    }
+    else {
+      def eligible(token: TokenInformation): Boolean = {
```

Comment:
Is it possible to avoid duplicating eligible()?

Reply:
hmm..not found elegant way to do this.  will try  to check any other approach.